python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -466,9 +488,27 @@ def filter_in_place(self, key, clear_existing: bool = False):
                 self.message_types = copy.deepcopy(self._original_message_types)
                 self.time_range = copy.deepcopy(self._original_time_range)
                 self.remove_invalid_p1_time = False
+                self.requested_source_ids = None
             else:
                 self.index = self._original_index
 
+        # Set requested source IDs.
+        if source_ids is not None:
+            source_ids = set(source_ids)
+            # Apply source IDs that are available.
+            if self.available_source_ids != source_ids:
+                unavailable_source_ids = list(source_ids.difference(self.available_source_ids))
+                if len(unavailable_source_ids) > 0:
+                    self.logger.warning('Not all source IDs requested are available. Cannot extract the following '
```

Comment:
I don't think this should be a warning. Maybe a debug print, but this is totally normal. The user's list is "this is what I would like _if available_", not "you must give me data for all of these".

Reply:
On second thought, right now we don't have a way to reset the source IDs filter when you call `filter_in_place()`. You can only clear all filters and then start over. If you pass in `None`, it gets ignored to avoid undoing one filter when you're changing another.

We should maybe treat an empty set as an explicit "allow all" instead.